libazureinit/src/provision/ssh.rs

```diff
@@ -29,10 +29,12 @@ pub(crate) fn provision_ssh(
     // the permissions are correct.
     std::fs::set_permissions(&ssh_dir, Permissions::from_mode(0o700))?;
 
-    let authorized_keys_path = match get_authorized_keys_path_from_sshd()? {
-        Some(path) => user.dir.join(path),
-        None => user.dir.join(".ssh/authorized_keys"),
-    };
+    let authorized_keys_path_string = authorized_keys_path_string
+        .unwrap_or_else(|| {
+            get_authorized_keys_path_from_sshd(get_sshd_g_output)
+                .unwrap_or_else(|| ".ssh/authorized_keys".to_string())
+        });
+    let authorized_keys_path = user.dir.join(authorized_keys_path_string);
```

Comment:
Just a minor suggestion.

```suggestion
    let authorized_keys_path =
        user.dir
            .join(authorized_keys_path_string.unwrap_or_else(|| {
                get_authorized_keys_path_from_sshd(get_sshd_g_output)
                    .unwrap_or_else(|| ".ssh/authorized_keys".to_string())
            }));
```
without having to redefine `authorized_keys_path_string`.

Reply:
good point! done, thanks!